janitor/functions/utils.py

```diff
@@ -217,7 +217,20 @@ def _computations_expand_grid(others: dict) -> pd.DataFrame:
     contents = {}
     for key, value, grid_index in grid:
         contents.update(_expand_grid(value, grid_index, key))
-    return pd.DataFrame(contents, copy=False)
+    # check length of keys and pad if necessary
+    lengths = set(map(len, contents))
+    if len(lengths) > 1:
+        lengths = max(lengths)
+        others = {}
+        for key, value in contents.items():
+            len_key = len(key)
+            if len_key < lengths:
+                padding = [""] * (lengths - len_key)
+                key = (*key, *padding)
+            others[key] = value
+    else:
+        others = contents
+    return others
```

Comment:
So `_computations_expand_grid` now returns a dict, not a dataframe right? Can you update the docstring and return type annotations accordingly?

ps: actually I didn't quite get the change on the return type, is there a specific reason why we're now returning a dict?

Reply:
with a dictionary i can build an index without having to go thru a dataframe - the previous setup returned a dataframe, from which I would then build an index (if exposing missing rows on the index)